Counter/src/app/shared/pokemon-service.ts

```diff
@@ -11,28 +12,34 @@ export class PokemonService {
     private filterStream: BehaviorSubject<string> = new BehaviorSubject("");
     public filter: Observable<string>;
     public pokemon: Observable<Pokemon[]>;
+    public pokemonList: Observable<Pokemon[]>;
     private filteredStream: BehaviorSubject<boolean> = new BehaviorSubject(false);
     public filtered: Observable<boolean>;
+    private userId: Observable<number>;
     private pokeUrl = 'api/pokemon';
     httpOptions = {
         headers: new HttpHeaders({ 'Content-Type': 'application/json' })
     };
 
-    constructor(private http:HttpClient) { 
-        this.pokemon = this.filterStream.pipe(
-            switchMap(searchTerm => {
-                if (!searchTerm) {
+    constructor(private http:HttpClient,
+        private userService: UserService) { 
+        this.userId = this.userService.getId();
```

Comment:
getId returns an `Observable<number>` which means it's asynchronous. It cannot be used right away, and I see now that you know that as well. The confusing part here is that you set the `getId` result to a variable `this.userId`. This is a bit dangerous because you might be tempted to use that as if it is the actual userId and not an obseservable. I recommend not using a variable and just putting in the userService reference when you use it later in the combineLatest.

Reply:
I removed all instances of where getId was used to initialize a variable in pokemon-service, hall-of-fame-service, and new-pokemon.